src/steamship/agents/llms/steamship_llm.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from pprint import pformat
+from typing import List, Optional
+
+from steamship import Block, File, MimeTypes, PluginInstance, Steamship, SteamshipError, Tag
+from steamship.agents.schema import Tool
+from steamship.data import GenerationTag, TagKind
+from steamship.plugin.capabilities import (
+    CapabilityPluginRequest,
+    CapabilityPluginResponse,
+    ConversationSupport,
+    FunctionCallingSupport,
+    SystemPromptSupport,
+)
+
+
+class SteamshipLLM:
+    """A class wrapping LLM plugins."""
+
+    def __init__(self, plugin_instance: PluginInstance):
+        # TODO (PR callout): In the spirit of getting folks to pin to specific instances more often, take the route of
+        #  using specific PluginInstances rather than client.use_plugin all the time with the params that we determine?
+        self.client = plugin_instance.client
+        self.plugin_instance = plugin_instance
+
+    @staticmethod
+    def with_gpt4(client: Steamship, temperature: float = 0.4, max_tokens: int = 256):
```

Comment:
**[Style, Nonblocking]**

I really like what this implies.. sort of:

```
SteamshipLLM.with_gpt4
SteamshipLLM.with_llama2
SteamshipLLM.with_vicuna

etc
```

A few thoughts:

* To the extent that there are common, known params, I think this is an opportunity to officially type and standardize them in the SDK (temperature, etc)
* The `with_` feels a bit weird to me.. I think because it's a Python reserved word used in a very specific way that isn't the way this is being used here. What if it was just: `LLM.gpt4()` or even `LLM.gpt()`
* I'm not sure if we need the `Steamship` prefix since that's already the scope of this entire library

Reply:
Re: `with_`, I'm emulating `steamship.agents.utils.with_llm`.  Prefixing these in *some* way is nice because:
- IDE users can type `.with_` and get a list of things they can use
- It avoids collisions with method names we might otherwise want to use in the case of someone naming their model in a generic way

Using `Steamship` as a prefix was a suggestion in the design proposal:
> Consolidate LLM and ChatLLM into single abstract base class (proposed name: SteamshipLLM to avoid collisions on updates).

Reply:
Instead of `with` and the python association, we could use: `using_` or `from_` or even `wrapping_` and get the same benefits. WDYT?

- `SteamshipLLM.using_openai_chatcomplete` or
- `SteamshipLMM.using_replicate_model` or
- `SteamshipLLM.from_openai` or
- `SteamshipLLM.wrapping_openai` or ...